beacon-chain/sync/initial-sync/blocks_fetcher.go

```diff
@@ -159,6 +160,27 @@ func newBlocksFetcher(ctx context.Context, cfg *blocksFetcherConfig) *blocksFetc
 	}
 }
 
+// This specifies the block batch limit the initial sync fetcher will use. In the event the user has provided
+// and excessive number, this is automatically lowered.
+func maxBatchLimit() int {
+	currLimit := flags.Get().BlockBatchLimit
+	maxLimit := params.BeaconConfig().MaxRequestBlocks
+	if params.DenebEnabled() {
```

Comment:
The limit is a function of which p2p topic we are subscribed to. I think we need to base this on current slot.

Reply:
Take a look at https://github.com/prysmaticlabs/prysm/blob/5a668079895e5cb7eef9d9102ce986c466c843fa/config/params/network_config.go#L47

Reply:
Resolved in side discussion - this is kind of a hack but the effect is only lowering the batch size. Larger batch sizes probably are not that helpful anyway given how we parallelize requests.